feat(client): reuse cashnotes for address payments

b-zee opened the pull request.

b-zee committed:
feat(client): reuse cashnotes for address payments

joshuef commented on sn_client/src/wallet.rs:
```diff
@@ -170,6 +170,26 @@ impl WalletClient {
+                let notes = self.wallet.get_payment_cash_notes(xor);
+                for note in notes {
+                    // Find a payment we're doing to an address, for which we have a cashnote already
+                    if let Some(e) = payments
```
Can we clarify `e` here? 

Also `payment we're doing for an address` feels confusing. Are we not looking for a given main pubkey for all payments _to_ an address? 

I think we can clarify the comment there a wee bit.

b-zee committed:
refactor: separate method and write test

b-zee committed:
feat(sn_transfers): impl From for NanoTokens

b-zee committed:
test(sn_transfers): add reuse cashnote cases

b-zee committed:
fix(sn_transfers): reuse payment overflow fix

When store cost reduces, the algorithm would use checked_sub, overflowing and discarding the output. This is now fixed, and it will also remove entries that have no content.

b-zee committed:
refactor(sn_transfers): clippy and fmt

b-zee committed:
Merge remote-tracking branch 'origin/main' into feat-reuse-previous-cashnotes
